packages/medusa/src/scripts/discount-rule-migration.ts

```diff
@@ -0,0 +1,144 @@
+import dotenv from "dotenv"
+import { createConnection, SelectQueryBuilder } from "typeorm"
+import Logger from "../loaders/logger"
+import {
+  DiscountCondition,
+  DiscountConditionOperator,
+  DiscountConditionType,
+} from "../models/discount-condition"
+import { DiscountConditionProduct } from "../models/discount-condition-product"
+import { DiscountRule } from "../models/discount-rule"
+dotenv.config()
+
+const typeormConfig = {
+  type: process.env.TYPEORM_CONNECTION,
+  url: process.env.TYPEORM_URL,
+  username: process.env.TYPEORM_USERNAME,
+  password: process.env.TYPEORM_PASSWORD,
+  database: process.env.TYPEORM_DATABASE,
+  migrations: [process.env.TYPEORM_MIGRATIONS as string],
+  entities: [process.env.TYPEORM_ENTITIES],
+  logging: true,
+}
+
+const migrate = async function ({ typeormConfig }): Promise<void> {
+  const connection = await createConnection(typeormConfig)
+
+  const BATCH_SIZE = 1000
+
+  await connection.transaction(async (manager) => {
+    const discountRuleCount = await manager
+      .createQueryBuilder()
+      .from(DiscountRule, "dr")
+      .getCount()
+
+    // INSERT INTO discount_condition(id, operator, type, discount_rule_id)
+    // (SELECT ROW_NUMBER() OVER(ORDER BY sq.d_id), sq.op, sq.tp, sq.d_id FROM
+    // 	(SELECT DISTINCT
+    // 			'in'::discount_condition_operator_enum op,
+    // 			'products'::discount_condition_type_enum tp,
+    // 			discount_rule_products.discount_rule_id d_id
+    // 	FROM discount_rule_products) sq)
+
+    let offset = 0
+    while (offset < discountRuleCount) {
+      const test = await manager
```

Comment:
nit: Fix naming  
```suggestion
      const discountRules = await manager
```

Change to be reflected on line 64